### PR TITLE
misc: added libconfig as a subproject

### PR DESCRIPTION
**Pull request by Monsterovich**

### CODE REVIEW

**Comment by yshui** [comment]
git submodule in general is a pain with work with, so I try to avoid them when possible. and I do want to push debian a little bit to update their libconfig. I am not objecting to adding this but, I would rather prefer either:

1. a meson wrap file, or
2. instructions in README about how to build libconfig.

**Comment by Monsterovich** [comment]
@yshui 

> 1. a meson wrap file, or

Done, I went for a master branch because tag v1.7.3 doesn't build, most likely due to a bug in the CMake files which they fixed later.

File: .gitignore [hunk 1]
```diff
@@ -72,3 +72,6 @@ doxygen/
 *.orig
 /tests/log
 /tests/testcases/__pycache__/
+
+# Subproject files
+subprojects/libconfig
```
Comment (absolutelynothelix) [inline .gitignore]: no newline at the end of the file

File: src/meson.build [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (absolutelynothelix) [inline src/meson.build]: why there's an empty line now?

File: subprojects/libconfig.wrap [hunk 3]
```diff
@@ -0,0 +1,4 @@
+[wrap-git]
+url = https://github.com/hyperrealm/libconfig
+revision = f9404f60a435aa06321f4ccd8357364dcb216d46
+depth = 1
```
Comment (absolutelynothelix) [inline subprojects/libconfig.wrap]: no newline at the end of the file

**Review by absolutelynothelix — COMMENTED** [review]
also it'd be good to squash all this merge commit madness. i personally think it could be a single commit but it's up to you (and yshui).

**Comment by Monsterovich** [comment]
> also it'd be good to squash all this merge commit madness. i personally think it could be a single commit but it's up to you (and yshui).

Done.

**Pull request merged**